MBBSEmu/HostProcess/MbbsHost.cs

```diff
@@ -1287,16 +1313,63 @@ private void DisableModule(string moduleId)
             _modules[moduleId].ModuleConfig.ModuleEnabled = false;
         }
 
+        private Timer SetupCleanupWarningTimer()
+        {
+            _cleanupWarningMinutesRemaining = CleanupWarningInitialMinutes;
+
+            var initialWarningTime = _cleanupTime + _cleanupGracePeriod - TimeSpan.FromMinutes(CleanupWarningInitialMinutes);
+            var dueTime = NowUntil(initialWarningTime);
+            var period = TimeSpan.FromMinutes(1);
+
+            Logger.Debug($"_cleanupTime Hours: {_cleanupTime.Hours}");
+            Logger.Debug($"_cleanupTime Minutes: {_cleanupTime.Minutes}");
+            Logger.Debug($"initialWarningTime Hours: {initialWarningTime.Hours}");
+            Logger.Debug($"initialWarningTime Minutes: {initialWarningTime.Minutes}");
+            Logger.Debug($"dueTime: {NowUntil(initialWarningTime).Minutes}");
+            Logger.Debug($"period: {period.Minutes}");
+
+            return new Timer(SendCleanupWarning, null, (int) dueTime.TotalMilliseconds, (int) period.TotalMilliseconds);
+        }
+
+        private void SendCleanupWarning(object _)
+        {
+            if (_cleanupWarningMinutesRemaining > 0)
+            {
+                Logger.Debug($"in SendCleanupWarning, _cleanupWarningMinutesRemaining = {_cleanupWarningMinutesRemaining}");
+
+                var minuteText = (_cleanupWarningMinutesRemaining > 1) ? "minutes" : "minute";
+
+                foreach (var c in _channelDictionary)
```

Comment:
You can combine the channel selection on the next line and the `foreach` definition into one line using a LINQ expression for the values being iterated over: 

`foreach (var channel in _channelDictionary.Select(c => c.Value.Channel))`